replication-api-impl/src/main/java/com/connexta/replication/api/impl/persistence/pojo/FilterIndexPojo.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.persistence.pojo;
+
+import com.connexta.replication.api.data.FilterIndex;
+import java.time.Instant;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.springframework.data.solr.core.mapping.Indexed;
+import org.springframework.data.solr.core.mapping.SolrDocument;
+
+/**
+ * This class provides a pojo implementation for a {@link
+ * com.connexta.replication.api.data.FilterIndex} capable of reloading all supported fields for all
+ * supported versions from the database. It also provides the capability of persisting back the
+ * fields based on the latest version format.
+ */
+@SolrDocument(collection = FilterIndexPojo.COLLECTION)
+public class FilterIndexPojo extends Pojo<FilterIndexPojo> {
+
+  /**
+   * List of possible versions:
+   *
+   * <ul>
+   *   <li>1 - initial version.
+   * </ul>
+   */
+  public static final int CURRENT_VERSION = 1;
+
+  public static final String COLLECTION = "filter_index";
+
+  @Nullable
+  @Indexed(name = "modified_since", searchable = false)
+  private Instant modifiedSince;
+
+  @Nullable
+  @Indexed(name = "filter_id")
+  private String filterId;
+
+  public FilterIndexPojo() {
+    super.setVersion(FilterIndexPojo.CURRENT_VERSION);
+  }
+
+  /**
+   * @see FilterIndex#getModifiedSince()
+   * @return the {@link Instant}, or {@code null} if no metadata has attempted to be replicated
+   */
+  @Nullable
+  public Instant getModifiedSince() {
+    return modifiedSince;
+  }
+
+  /**
+   * @param modifiedSince the modified since time
+   * @return this pojo
+   */
+  public FilterIndexPojo setModifiedSince(Instant modifiedSince) {
+    this.modifiedSince = modifiedSince;
+    return this;
+  }
+
+  /** @return the filter id for this this index belongs to */
+  @Nullable
+  public String getFilterId() {
```

Comment:
❓ Did we consider making this the actual id for the index as opposed to a field? This would ensure that only one object would ever exist for a given filter. Also it would make it easier for the filter removal to also remove the corresponding index which it should. So please coordinate with @kcover As removing the filter config should remove this object as well.

Reply:
I still do not like having two persistent objects sharing the same UUID. I'd rather have a specific `removeByFilterId` on the FilterIndexRepository/Manager class to simplify removal of the FilterIndex.

Reply:
Unless you can guarantee that no 2 objects with different id but the same filter id can exist, I see no other ways. That check cannot be done in the code.

Reply:
Left a todo for now until Kiefer's PR is merged.